### ISSUE TITLE
Need help sorting out docker and IN issues

### ISSUE THREAD

**Issue by ovizii** [invoiceninja/dockerfiles#182]
I have been happily running IN for a couple of years on docker with docker-compose. Unfortunately, when I started using it I knew little about docker so I was pulling this image: image: invoiceninja/invoiceninja without a tag

![image](https://user-images.githubusercontent.com/837902/89770176-adc2e200-dafe-11ea-9f47-96c2fb75a16f.png)

today I updated and things broke as the latest image as I found out has heavily changed. I then looked into the latest image I had previously pulled without a tag and it said 4.5.17

so I now went back and use image: invoiceninja/invoiceninja:4.5.17 and am getting back my dashboard but still have the following problems:

**in my dashboard I see:**

**A new version of Invoice Ninja is available. You’re running v4.5.16, the latest is v4.5.17**
this is nonsense, my docker-compose states:
app:
image: invoiceninja/invoiceninja:4.5.17

I also just bought the white label extension and the dashboard still says:
**Your white label license has expired, please consider renewing it to help support our project.**

Can somebody please explain the way forward so that my IN instance gets to a stable and updated version while staying usable?

**I kinda need an upgrade path and how to proceed please.** Please keep in mind that I run a lot of containers on this host, all behind traefik as reverse proxy which sorts out all the certificates so I need to pass my requests through traefik to (ideally) a plain http port where IN is listening.

Reply

**Comment by codedge** [comment]
Hey @ovizii 

I guess you should have a look at the [IN setup with nginx](https://github.com/invoiceninja/dockerfiles/wiki/docker-compose-with-nginx). This comes without any TLS stuff.

There is a guide for version 4 - main thing is that you have the `image` `invoiceninja/invoiceninja:alpine-4`. This autmatically has the most recent version (which is 4.5.19 currently).

Why you still have this issue with "A new version of Invoice Ninja is available. You’re running v4.5.16, the latest is v4.5.17" I cannot solve. The same applies to the white label issue. I am pretty sure @turbo124 can help you with both.

**Comment by turbo124** [comment]
@ovizii send us an email to contact@invoiceninja.com and we will help with the whitelabel issue.

The app does require outbound internet access to validate the white label license, i wonder if your setup may be preventing this?

**Comment by ovizii** [comment]
hi guys,

thanks for the replies but I haven't gotten around to sort this out and before I start following these instructions, is there maybe a clean upgrade path from my v4.5.17 to the latest IN version? (Please note that it would need to be run behind a reverse proxy)

**Comment by turbo124** [comment]
@ovizii V4 and V4 are completely different codebases and schema's, however you are able to migrate your data from V4 to V5 using the migration too in V4 under Settings > Account Management.

**Comment by ovizii** [comment]
@turbo124  you can safely close this but could you give me more details about how to migrate my data? I can't seem to find that option, here is waht I see at Settings => Account Management

![image](https://user-images.githubusercontent.com/837902/103649281-7f86e880-4f5e-11eb-9d69-2a656fdcc8d2.png)

ah, I think I have found it, is this what you are referring to? (Import / Export)

![image](https://user-images.githubusercontent.com/837902/103649437-bd840c80-4f5e-11eb-904a-1932e99f98d3.png)

**Comment by ovizii** [comment]
@turbo124  

indeed my instance uses:

image: invoiceninja/invoiceninja:4.5.17

and the last version 4 seems to be 4.5.18 - is that the one that offers the feature you mentioned?

**Comment by ovizii** [comment]
YES! 4.5.18 had the feature you mentioned!

![image](https://user-images.githubusercontent.com/837902/103872275-7a48ab80-50ce-11eb-81d4-f59adf9d83ff.png)

**Comment by turbo124** [comment]
Hi @ovizii you may want to wind forward the version to the latest of V4 as there are significant changes. I am pretty sure the migration will fail using 4.5.18 due to bugs in that very of the migration.

**Comment by ovizii** [comment]
@turbo124 I would if I knew how. Sorry to bug you in this thread but I haven't found the information elsewhere and you seem very knowledgeable about this matter. 

I checked the last few pages of the tags page sorted by name and the last is 4.5.18 => https://hub.docker.com/r/invoiceninja/invoiceninja/tags?page=4&ordering=-name 

Yes, there are others but I don't know how to upgrade from say 

invoiceninja/invoiceninja:4.5.18 

to something like:

invoiceninja/invoiceninja:alpine-4.5.26

Would an in-place-upgrade like this work?

**Comment by turbo124** [comment]
that yes, that is correct. alpha-4.5.26 will work.

**Comment by ovizii** [comment]
@turbo124 
Since I can't find alpha-4.5.26 - I assume that was autocorrect and you meant to type alpine-4.5.26 ?

**Comment by turbo124** [comment]
sorry, typo :) you are correct.

**Comment by ovizii** [comment]
I guess I'll have to try my luck with 4.5.18 - the update to alpine-4.5.26 gave me a handful of errors like 

```
Creating invoiceninja_web  ... done
t found in $PATH": unknown'

ERROR: for cron  Cannot start service cron: b'OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \\"bash\\": executable file not found in $PATH": unknown'
ERROR: Encountered errors while bringing up the project.
```

and the app and the cron containers are constantly restarting. Might give it another try this weekend. 


Thanks for all the help and info, its very much appreciated.

**Comment by ovizii** [comment]
@turbo124 - I still gave it a try and it seems there's a permission problem between version 4.5.18 and alpine-4.5.26

```
chown: /var/www/app/storage/framework/sessions/cAay0GOHcyRGICSlTIMqsnjCgIQPy20GZfwSMxaI: Operation not permitted                                                                │
│   chown: /var/www/app/storage/framework/sessions/lZgvrRCJiI9duxJZwwW34vS2f3w9YdDcvGg6yB9I: Operation not permitted                                                                │
│   chown: /var/www/app/storage/framework/sessions/jNAEYzlYk85qbBU9sqBkVW53AbEUjB4J3YAwMvep: Operation not permitted                                                                │
│   chown: /var/www/app/storage/framework/sessions/bK6aIHZWVXH2b556PWzCKNOAI40y6y9x60qlTPNg: Operation not permitted                                                                │
│   chown: /var/www/app/storage/framework/sessions: Operation not permitted                                                                                                         │
│   chown: /var/www/app/storage/framework/sessions: Operation not permitted                                                                                                         │
│   chown: /var/www/app/storage/framework: Operation not permitted                                                                                                                  │
│   chown: /var/www/app/storage/framework: Operation not permitted                                                                                                                  │
│   chown: /var/www/app/storage: Operation not permitted                                                                                                                            │
│   chown: /var/www/app/storage: Operation not permitted
```

the volume mounted into invoiceninja was owned by www-data:www-data and permissions were 770 - any hint where I could read up on changes which happened in this alpine-4.5.26? Was the owner changed to something different from www-data? This way my app container isn't starting :-(

**Comment by turbo124** [comment]
so you'll want to bring the container down and then

```sudo chown -R 1500:1550 docker/app/```

and then bring it back up again.

**Comment by ovizii** [comment]
Thanks that did the trick - all working now except for the cron container. That simply says: 

`ERROR: for invoiceninja_cron  Cannot start service cron: b'OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \\"bash\\": executable file not found in $PATH": unknown'`

Here is the corresponding docker-compose.yml part:

```
  cron:
    image: invoiceninja/invoiceninja:alpine-4.5.26
    container_name: invoiceninja_cron
    restart: unless-stopped
    links:
      - db:mysql
    env_file: .env
    volumes_from:
      - app
    entrypoint: |
      bash -c 'bash -s <<EOF
      trap "break;exit" SIGHUP SIGINT SIGTERM
      sleep 300s
      while /bin/true; do
        ./artisan ninja:send-invoices
        ./artisan ninja:send-reminders
        sleep 1h
      done
      EOF'
```

oh, I made a lucky guess and replaced all instances of "bash" with "sh" and it all works now. 

basically:
`sh -c 'sh -s <<EOF`

**Comment by ovizii** [comment]
@turbo124 
last one: 
I am still seeing this notice about an expired white label license despite having bought it in August 2020
![image](https://user-images.githubusercontent.com/837902/103882236-cb12d100-50db-11eb-84be-b67802d1461e.png)

I have the invoice with the license key, how do I solve this?

**Comment by turbo124** [comment]
hey @ovizii  please send an email to contact@invoiceninja.com and we'll look into that for yo.

**Comment by ovizii** [comment]
Just to add to the saga, after it seemed that alpine-4.5.26 was working I realized PDF attachments are broken with both local phantomjs and phantomcloudjs. Upgrading to alpine-4 which currently means version 4.5.30 its still broken. Not sure if there is a relation to this thread here but I thought I'd post it. Here is the other forum post anyway. https://forum.invoiceninja.com/t/pdf-attachments-are-broken/5870
